### ISSUE TITLE
关于版本问题

### ISSUE THREAD

**Issue by kangyiwen** [kelaberetiv/TagUI#1212]
@ruthtxh ，你好，后面要多请教你技术问题了。 我提问是中文的，可以用英语回答，我自己谷歌翻译，谢谢。

我根据 https://tagui.readthedocs.io/en/latest/setup.html  安装后，TagUI 显示 v6.46，可以正常运行 click 等命令，但是无法运行excel读取。

我下载 https://github.com/kelaberetiv/TagUI/archive/master.zip 后覆盖本地文件，可以运行excel读取了，但版本显示是 v6.109

请问，在 https://github.com/kelaberetiv/TagUI/archive/master.zip 下载并覆盖，这个做法正确吗？ 如果正确，为什么版本还从v6.46倒退到 v6.109 了？

**Comment by ruthtxh** [comment]
Hi @kangyiwen 

v6.46 is the major release version that we have packaged for installation. There were some bug fixes and new features like the Excel integration that were added along the way after v.6.46, hence the newer versions. The quickest way to upgrade your TagUI is to run `tagui update` from your command prompt. Downloading from the link you provided and copy/pasting over your src folder works too.

**Comment by kangyiwen** [comment]
谢谢，明白了

https://github.com/kelaberetiv/TagUI/archive/refs/heads/master.zip
https://github.com/kelaberetiv/TagUI/archive/master.zip

这2个路径下载的，应该是一样的吧？

**Comment by kangyiwen** [comment]
我折腾读取 excel，弄了2天，各种弄，都没搞定，最终原来是版本内不含这个命令。

希望可以做一个新版本的 EXE 安装文件，这样最方便新用户。

**Comment by kensoh** [comment]
Adding on, oh the `tagui update` step is part of the installation steps at https://tagui.readthedocs.io/en/latest/setup.html

It is at step 8. Maybe we can change the way it is written so that users can see that step clearly? Your thoughts Ruth?

We didn't package a .exe every time there is an update because there has been frequent updates and it takes significant time to generate the .exe (for Windows) and the .zip files (for Windows, Mac, Linux). Thus, due to resource limitation, we include the `tagui update` step as part of the installation instructions, so that users always have the latest version.

**Comment by kangyiwen** [comment]
谢谢，在安装说明了更清晰的表达，会减少用户一些困扰，一部分用户会因为无法弄明白某些语法报错原因，而放弃使用。

tagui update 在我的WINDOWS上报错 ERROR - cannot find updata  我也没找是什么原因。

**Comment by kangyiwen** [comment]
![image](https://user-images.githubusercontent.com/76672577/165135110-81950c50-f9f8-4ba7-b847-c1d4d1e1b3cb.png)
抱歉，上面是我打错字了。 实际使用是 tagui updata。

**Comment by kangyiwen** [comment]
@ruthtxh @kensoh 这里可能是错别字，见图：

https://tagui.readthedocs.io/en/latest/reference.html#read
![111](https://user-images.githubusercontent.com/76672577/165200431-1227b3b9-fb40-4e75-86b6-4743bdabc3e8.png)

**Comment by kangyiwen** [comment]
@ruthtxh 你好，请问Github下载的TagUI版本并不是v6.87，这里提到的 v6.87 请问在哪里下载？   
https://tagui.readthedocs.io/en/latest/reference.html#read
![image](https://user-images.githubusercontent.com/76672577/165200699-05ad7183-2f96-4350-b67c-45cbec917f94.png)
谢谢

**Comment by ruthtxh** [comment]
Hi @kangyiwen 

The documentation is correct.

You will get the latest version v6.109 when you update, which is a later release than v6.87. Hence that bug fix is already included.

**Comment by kangyiwen** [comment]
哦！！！   对 对，是我看理解错版本了，脑子里把   v6.87 理解成 6.8.7， v6.109理解成v6.1.0.9。
很明显，我开发经验不足啊  ：）

**Comment by kensoh** [comment]
Good idea @ruthtxh, you can try to make the modifications in `docs/setup.rst` and commit.

If there's some opportunity to iterate, I can make further commit. The .rst syntax is relatively new to me too (and quite confusing the syntax), but I use the existing .rst examples from Yi Sheng to follow and make modifications.

**Comment by ruthtxh** [comment]
@kensoh ok I've updated the docs for tagui update to be in code block instead

**Comment by kangyiwen** [comment]
我用 tagui update ，报错：`ERROR - internet connection to access GitHub is required to update TagUI`

看来在大陆，用 Github 还是挺麻烦的事情啊。

**Comment by ruthtxh** [comment]
Hi @kangyiwen, yes internet connection is needed to pull the latest source code from GitHub. In that case you will have to manually download the files copy/paste to overwrite your src folder.

**Comment by kensoh** [comment]
This sounds like an opportunity for Yi Wen to maintain a forked version of TagUI that points to his website or somewhere else which can be reliably accessed in China. The code to modify will be in tagui/src/tagui and tagui/src/tagui.cmd file. Just search for `https://raw.githubusercontent.com` and change that URL to the new host.

**Comment by kangyiwen** [comment]
我购买了域名，也有闲置的服务器，明天zoom我们可以多聊聊。

**Comment by ruthtxh** [comment]
Currently we have directions to work on creating a new release to package the latest version, so this might solve this issue.

**Comment by kensoh** [comment]
Oh that'll be good news for user convenience. And probably make sense since there isn't frequent bug fixes needed, or frequent iteration to TagUI core. Let me know if you run into any trouble! The best way I found is to unzip an existing zip for an OS and overwrite the files with master.zip, then zip back. This minimise local artifacts and temp files, and also deal with situations like the jython jar file goes missing on local TagUI copy. Also, to preserve OS specific attributes, this has to be done on respective OSes.

**Comment by kangyiwen** [comment]
@ruthtxh 真是太好了！ 请问大约多久，可以有新版本的 EXE 安装文件?

**Comment by ruthtxh** [comment]
Hi @kangyiwen we will target the new packaged release in 1 month's time.
